src/text-editor.coffee

```diff
@@ -2952,7 +2952,13 @@ class TextEditor extends Model
 
   # TODO: Rename to foldRowRange?
   createFold: (startRow, endRow) ->
-    @displayBuffer.createFold(startRow, endRow)
+    fold = @displayBuffer.createFold(startRow, endRow)
+    # Update the buffer positions based on the screen positions.
+    # For example, the end of the screen fold line maps to the end of the buffer fold.
+    # @moveCursors (cursor) ->
```

Comment:
I would like to do update the cursors so the screen position stays in sync with the buffer position.
Let me know if this would be useful.

This breaks some tests though and I can't seem to fix some of them due to #10571:
- The fold creation would update the cursors, changing selections inside the fold.
- Adding the selections afterwards would remove the fold.